### ISSUE TITLE
Inconsistent CSS positions for components between the dev and node adapter

### ISSUE THREAD

**Issue by stalkerg** [sveltejs/kit#7426]
### Describe the bug

I have global styles in the `<svelte:head>` for the main +layout.svelte.
On the dev env, the components `<styles>` put after main `<svelte:head>` but [adapter-node](https://github.com/sveltejs/kit/blob/master/packages/adapter-node) inject it before  `<svelte:head>`. As you can understand, style and behavior depend on order, and some of my rules are not working as I expected. 

### Reproduction

Compare CSS position for dev and adapter-node relative `<svelte:head>`.

### System Info

```Shell
System:
    OS: Linux 5.19 Gentoo Linux
    CPU: (24) x64 AMD Ryzen 9 5900X 12-Core Processor
    Memory: 5.70 GB / 31.26 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.11.0 - /usr/bin/node
    npm: 8.19.2 - /usr/bin/npm
  npmPackages:
    @sveltejs/adapter-node: 1.0.0-next.99 => 1.0.0-next.99 
    @sveltejs/adapter-static: 1.0.0-next.46 => 1.0.0-next.46 
    @sveltejs/kit: 1.0.0-next.526 => 1.0.0-next.526 
    @sveltejs/svelte-virtual-list: github:sveltejs/svelte-virtual-list => 3.0.1 
    svelte: 3.52.0 => 3.52.0 
    vite: ^3.2.1 => 3.2.1
```

### Severity

blocking all usage of SvelteKit

**Comment by stalkerg** [comment]
Node adapter:
![Screenshot_20221029_100817](https://user-images.githubusercontent.com/1853716/198755088-76ae8ebd-8a0c-4f93-92f0-f1d5e9e55f17.png)

dev
![image](https://user-images.githubusercontent.com/1853716/198755114-3a5c591d-4c26-4608-ba11-0619781ba624.png)

**Comment by Rich-Harris** [comment]
This will need a minimal reproduction if we're to create a test case and possible fix

**Comment by stalkerg** [comment]
@Rich-Harris okey, reproducing is very trivial:
1. install sveltkit starter demo, `npm create svelte@latest my-app`
2. run it in dev `npm run dev`, and as you can see, all CSS after `<title>`:
![изображение](https://user-images.githubusercontent.com/1853716/199835885-a512117f-d065-49c2-a0b9-219d3bd94a4c.png)
3. run it as preview `npm run preview`, and as you can see, all CSS before `<title>`:
![изображение](https://user-images.githubusercontent.com/1853716/199836193-726d1386-258c-485d-a96a-7bd802581ee7.png)

**Comment by Rich-Harris** [comment]
This shows tags coming before/after the `<title>`, but to build a test case around this the styles themselves need to differ between prod and dev

**Comment by stalkerg** [comment]
We just need to add any `<link` to css in `<svelte:head>` and this will be illustrated, depending on before/after some rules will be overridden each over or not.
I am currently moving into a new house and can probably make a unit/integration test only next week.

**Comment by rodshtein** [comment]
![Screen Recording 2022-11-07 at 5 18 00 PM](https://user-images.githubusercontent.com/13717428/200335985-a6cfe910-033e-419e-9c09-dac58531e999.gif)
On preview above: right window is `preview` mode, left is `dev`

@Rich-Harris i made an simple reproduction 
https://stackblitz.com/edit/sveltejs-kit-template-default-g3jmhk?file=src/routes/+page.svelte

Main points:
- `+layout` has imported styles for `<h1>`  
- `+page` has its own styles for `<h1>`
-  on `dev mode`, imported styles in `+layout` have priority over page styles
-  but after a hot update, the `+page` styles override the styles imported into `+layout`.

**Comment by stalkerg** [comment]
In my opinion, the component CSS styles should have a chance to overwrite any styles from `<svelte:head>` and `+layout`.

**Comment by stalkerg** [comment]
hmm it's a huge stopper, in that case not possible to use any external CSS with SvelteKit especially if you must overwrite something.

**Comment by Rich-Harris** [comment]
Does this issue only happen if you're doing stuff in `<svelte:head>`? Because if so my best advice would be 'don't do that'

**Comment by stalkerg** [comment]
Yes, but how to support dynamic external CSS? The URL depends on the current user's theme choice. 
```
<link href="{env.style_cdn_host}/{env.site_theme}.css?v=144" rel="stylesheet" type="text/css" />
```
I think I can implement by `%mycssurl%` replacement in `app.html` but in that case, a user must reload the full page, as I understand.

**Comment by zyxd** [comment]
> Yes, but how to support dynamic external CSS? The URL depends on the current user's theme choice.
> 
> ```
> <link href="{env.style_cdn_host}/{env.site_theme}.css?v=144" rel="stylesheet" type="text/css" />
> ```
> 
> I think I can implement by `%mycssurl%` replacement in `app.html` but in that case, a user must reload the full page, as I understand.

This doesn't solve the problem in this topic, but for switching CSS themes loaded dynamically from external sources, you can do all the work in `<head>` rather than in the `<style>` of the component.

I mean, those styles that can be overwritten in the dynamically loaded theme should also be loaded via `head > link`.

https://stackblitz.com/edit/sveltejs-kit-template-default-ikbt4z

**Comment by Rich-Harris** [comment]
> Yes, but how to support dynamic external CSS?

You could do it imperatively — add this to `src/app.html`...

```html
<link id="theme" rel="stylesheet" href="REPLACE_ME_IN_HANDLE_HOOK">
```

...then do this sort of thing:

```js
$: document.querySelector('theme').href = `${style_cdn_host}/${site_theme}.css?v=144`;
```

That way, you have complete control over where in the `<head>` the stylesheet goes, and with it the order in which stylesheets are applied

**Comment by stalkerg** [comment]
@Rich-Harris  unfortunately it's not working for SSR because you have no access to `document`.

**Comment by stalkerg** [comment]
@Rich-Harris seems like working BUT looks awkward especially because I should share session data between `+layout.server.js` and `hooks.server.js` by a global variable (also it's probably can be the point of race condition).
